code/modules/mob/living/carbon/life.dm

```diff
@@ -236,21 +235,18 @@
 		wetlevel = max(wetlevel - 1,0)
 
 /mob/living/carbon/handle_stomach()
-	spawn(0)
-		for(var/mob/living/M in stomach_contents)
-			if(M.loc != src)
+	for(var/mob/living/M in stomach_contents)
+		if(M.loc != src)
+			stomach_contents.Remove(M)
+			continue
+		if(stat != DEAD)
+			if(M.stat == DEAD)
 				stomach_contents.Remove(M)
+				qdel(M)
 				continue
-			if(istype(M, /mob/living) && stat != 2)
-				if(M.stat == 2)
-					M.death(1)
```

Comment:
Wait does this do a check for if someone's dead only on mobs that aren't dead? 😆